pandas/core/window.py

```diff
@@ -204,8 +204,47 @@ def __repr__(self):
         )
 
     def __iter__(self):
-        url = "https://github.com/pandas-dev/pandas/issues/11704"
-        raise NotImplementedError("See issue #11704 {url}".format(url=url))
+        closed = self.closed
+        window = self._get_window()
+        minp = _use_window(self.min_periods, window)
+        offset = _offset(window, self.center)
+
+        blocks, obj, index = self._create_blocks()
+        _, indexi = self._get_index(index)
+
+        for values in blocks:
+            arr = np.asarray(values)
+
+            start, end, N, win, _minp, is_variable = libwindow.get_window_indexer(
+                arr, window, minp, indexi, closed
+            )
+
+            if arr.ndim == 1:
+                arr = np.expand_dims(arr, axis=1)
+
+            counts = libwindow.roll_sum(
```

Comment:
Instead of this, could you just generate the slice and yield if the number non-null elements meets `_minp`?

```
result = values.iloc[slice(s, e)]
if result.count() > _minp:
    yield result
```